From a6722e167a03b749a07bb59f385fc378ddb659a2 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 19 Jun 2008 14:54:59 +0800
Subject: [PATCH] ENGR00072896 Power off function is not implemented in mx31

Assign mc13783 power off function to pm_power_off hook

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 arch/arm/mach-mx3/mx3_3stack.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx3/mx3_3stack.c b/arch/arm/mach-mx3/mx3_3stack.c
index 3a0a1ca..855db4e 100644
--- a/arch/arm/mach-mx3/mx3_3stack.c
+++ b/arch/arm/mach-mx3/mx3_3stack.c
@@ -53,6 +53,7 @@
 #include <asm/arch/gpio.h>
 #include <asm/arch/mmc.h>
 #include <asm/arch/spba.h>
+#include <asm/arch/pmic_power.h>
 
 #include "board-mx3_3stack.h"
 #include "crm_regs.h"
@@ -860,6 +861,9 @@ static void __init mxc_board_init(void)
 	mxc_init_mmc();
 	mxc_init_ide();
 	mxc_init_pata();
+
+	/* set power off hook to mc13783 power off */
+	pm_power_off = pmic_power_off;
 }
 
 #define PLL_PCTL_REG(pd, mfd, mfi, mfn)		\
-- 
1.5.4.4

